Refactors YamlJsonToggle component into a generic Toggle component

mabashian opened the pull request:
##### SUMMARY
I'd like to use this in the schedule details view for toggling the dates between local and utc timezones.

See this screenshot: https://tower-mockups.testing.ansible.com/patternfly/templates/templates-schedules-detail/

If/when something like this: https://github.com/patternfly/patternfly-react/issues/3847 is available in PF then we can start using that.

mabashian committed:
Refactors YamlJsonToggle component into something a little more generic so that it can be used to toggle between local and utc times in the schedule details view.

keithjgrant commented:
From our discussion offline about reducing the number of props on the Toggle, I came up with this. Maybe debatable whether this is "better", but it is an option:

```
<Toggle
  currentMode={mode}
  modes={[[YAML_MODE, 'YAML'], [JSON_MODE, 'JSON']]}
  onChange={newMode => {
    try {
      setCurrentValue(getValueAsMode(currentValue, newMode));
      setMode(newMode);
    } catch (err) {
      setError(err);
    }
  }}
/>
```

(perhaps calling the prop `options` rather than `modes`?)

and updating Toggle to render:

```
    <ButtonGroup>
      {modes.map(([value, label]) => (
        <SmallButton
          onClick={() => setValue(value)}
          variant={currentMode === value ? 'primary' : 'secondary'}
        >
          {label}
        </SmallButton>
      ))}
    </ButtonGroup>
```

This is maybe a slightly less obvious prop structure, but it does make the component a little more versatile in case we need a three-button Toggle in the future

mabashian committed:
Changes Toggle to MultiButtonToggle in an attempt to differentiate it from an upstream PF component.  Altered props to be a bit more concise as well as support more than two buttons.

mabashian commented:
@keithjgrant - took a stab at some of the changes your proposed here: https://github.com/ansible/awx/pull/6130/commits/1f0acef844203eac826d6db391a907b9277d986e

Highlights:

1) Consolidated `leftValue, leftLabel, rightValue, rightLabel` down to a single prop `buttons` which matches your proposed data structure.  I'm still a little fuzzy on the name.  Maybe `buttonOptions`?
  - This required writing a custom prop type validator to attempt to ensure that the correct data structure is passed in.  Is there an advantage to passing an array of arrays over an array of objects?  Something like `[{value: someValue, label: 'YAML'}, ...]`.  I ran into some issues with the linter on a [custom prop type validator that's required](https://github.com/yannickcr/eslint-plugin-react/issues/1020).  My solution is built off of https://github.com/facebook/react/issues/9125#issuecomment-285531461.
2) I changed the `currentMode` prop to `currentValue`.  Thinking about this component past just the yaml/json or local/utc toggles made me think that `value` may be more generic but I'm open to changing that back.
3) Changed the component name from `Toggle` to `MultiButtonToggle` to try to differentiate it from an upstream PF component.

keithjgrant commented on awx/ui_next/src/components/MultiButtonToggle/MultiButtonToggle.jsx:
```diff
@@ -0,0 +1,66 @@
+  font-size: var(--pf-global--FontSize--xs);
+`;
+
+function MultiButtonToggle({ buttons, currentValue, onChange }) {
```
I'd probably go with `value` as the prop name over `currentValue`, just because that's a common enough convention in other components

keithjgrant commented on awx/ui_next/src/components/MultiButtonToggle/MultiButtonToggle.jsx:
```diff
@@ -0,0 +1,66 @@
+  );
+}
+
+const buttonsPropType = {
```
Nice. I'm actually surprised prop-types doesn't already have a tuple type or something that can do this for is, but I couldn't find it either.

mabashian committed:
Change currentValue prop to simply value.  Adds basic unit test coverage to MultiButtonToggle component.
